File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/ledger/GetRevocRegDefRequestTest.java
package org.hyperledger.indy.sdk.ledger;

import org.json.JSONObject;
import org.junit.Test;

public class GetRevocRegDefRequestTest extends LedgerIntegrationTest {

	@Test
	public void testBuildGetRevocRegDefRequestWorks() throws Exception {
		JSONObject expectedResult = new JSONObject()
				.put("operation", new JSONObject()
						.put("type", "115")
						.put("id", revRegDefId)
				);

		String request = Ledger.buildGetRevocRegDefRequest(DID, revRegDefId).get();

		assert (new JSONObject(request).toMap().entrySet()
				.containsAll(
						expectedResult.toMap().entrySet()));
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/LedgerTests/GetRevocRegDefRequestTest.cs
using Hyperledger.Indy.LedgerApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.LedgerTests
{
    [TestClass]
    public class GetRevocRegDefRequestTest : IndyIntegrationTestWithPoolAndSingleWallet
    {
        [TestMethod]
        public async Task TestBuildGetRevocRegDefRequestWorks()
        {
            var expectedResult = "\"operation\":{\"type\":\"115\",\"id\":\"RevocRegID\"}";

            var request = await Ledger.BuildGetRevocRegDefRequestAsync(DID, "RevocRegID");

            Assert.IsTrue(request.Replace("\\s+", "").Contains(expectedResult.Replace("\\s+", "")));
        }
    }
}
